packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts

```diff
@@ -371,102 +401,382 @@ describe('Appointment Popup Form', () => {
   });
 
   describe('Recurrence', () => {
-    it.skip('Recurrence form should work properly if recurrenceRule property mapped recurrenceRuleExpr', async () => {
-      const { scheduler, POM } = await createScheduler({
-        dataSource: [{
-          text: 'Watercolor Landscape',
+    it('Recurrence editor container should be visible after changing its visibility value', async () => {
+      const appointment = {
+        text: 'Test Appointment',
+        startDate: new Date(2017, 4, 1, 9, 30),
+        endDate: new Date(2017, 4, 1, 11),
+      };
+
+      const { POM, scheduler } = await createScheduler(getDefaultConfig());
+
+      scheduler.showAppointmentPopup(appointment);
+
+      const recurrenceGroup = $(POM.popup.recurrenceGroup);
+
+      expect(recurrenceGroup.hasClass('dx-scheduler-form-recurrence-hidden')).toBe(true);
+
+      POM.popup.selectRepeatValue('weekly');
+
+      expect(recurrenceGroup.hasClass('dx-scheduler-form-recurrence-hidden')).toBe(false);
+    });
+
+    it('Check that after opening recurrence appointment current form is main form', async () => {
+      const appointment = {
+        text: 'Recurrent Appointment',
+        startDate: new Date(2017, 4, 1, 9, 30),
+        endDate: new Date(2017, 4, 1, 11),
+        recurrenceRule: 'FREQ=WEEKLY;BYDAY=MO,WE,FR;COUNT=10',
+      };
+
+      const { POM, scheduler } = await createScheduler(getDefaultConfig());
+
+      scheduler.showAppointmentPopup(appointment);
+
+      POM.popup.getEditSeriesButton().click();
+
+      const recurrenceGroup = $(POM.popup.recurrenceGroup);
+
+      expect(recurrenceGroup.hasClass('dx-scheduler-form-recurrence-hidden')).toBe(true);
```

Comment:
Suggestion: it's better to create a function at Popup's POM like `getVisibleFormGroupName` that returns 'main' or 'recurrence' depending on which form group is visible